docs/gallery/advanced_io/plot_editing.py

```diff
@@ -0,0 +1,178 @@
+"""
+Editing NWB files
+=================
```

Comment:
The title of the tutorial is a bit misleading. "Editing NWB files" suggests that this is a regular process of modifying an existing file, but I think that is not what this tutorial is targeting. It seems this tutorial targets more advanced manual surgeries. For clarity, I think we should distinguish between "Modifying an existing NWB file with PyNWB" and "Manually modifying an NWB file with h5py".

Reply:
Sure, that sounds good. We should add material on how to edit files using PyNWB. I am not sure what the capabilities of that are.

Reply:
Aside from renaming groups via `h5py.move`, I think the other items that are currently listed here are possible through PyNWB:

- For changes to h5py.datasets (e.g., changing values etc.), the PyNWB exposes the `h5py.Dataset` object directly to the user on read. I.e., if you open with `mode='a'` then any operations you do manually on f["acquisition"]["synthetic_timeseries"]["data"] in the tutorial, you should be able to do with `nwbfile.get_acquisition("synthetic_timeseries").data`. The same kind of risks of breaking things apply, but the access is through PyNWB. 
- For changing attribute values, I'm not 100% sure, but I believe you can change attribute values in PyNWB and then call write to update the file. The difference between attributes and datasets is that the datasets are updated immediately whereas attributes require an additional call to write. This is due to the fact that attributes don't have lazy read.
- For replacing a dataset, I believe we can use `export`. I.e., 1) open the file, 2) remove the `TimeSeries`, 3) add a new `TimeSeries` with the new data, 4) call export to write the updated file to a new location.